### PR TITLE
Add .vscode to .gitignore file and seperated that of macos with vscode

### LINKED ISSUE #291: [Feature] Add .vscode to .gitignore file
## Problem
Since the `.vscode` folder often contains settings to your local development workspace such as Extensions you have, custom key bindings, themes, and so on, It's best this file is ignored, to avoid conflicting with others

One way around is for each contributor to delete their `.vscode` folder if it exists, but it's easier just ignoring it with .gitinore

## Implementation
- [x] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by Vanessa082**
## Fixes
- Fixes #291  by @Vanessa082

## Description

Since the `.vscode` folder often contains settings to your local development workspace such as Extensions you have, custom key bindings, themes, and so on, It's best this file is ignored, to avoid conflicting with others

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,7 @@
-.DS_Store
\ No newline at end of file
+# macOS general
+# https://github.com/github/gitignore/blob/main/Global/macOS.gitignore
+.DS_Store
+
+# Visual Studio Code (VS Code)
+# https://github.com/github/gitignore/blob/main/Global/VisualStudioCode.gitignore
+.vscode/*
```
Comment (possumbilities) [inline .gitignore]: file is missing an End of File character

File: .vscode/settings.json
Comment (possumbilities) [inline .vscode/settings.json:1]: this file should not be a part of this PR

File: docs/index.html
Comment (possumbilities) [inline docs/index.html:1]: this file should not be a part of this PR

File: docs/style.css
Comment (possumbilities) [inline docs/style.css:1]: this file should not be a part of this PR

File: .vscode/settings.json
Comment (Vanessa082) [reply .vscode/settings.json:1]: I updated the commit, and removed the changes I made here 😊

File: docs/index.html
Comment (Vanessa082) [reply docs/index.html:1]: here too 😊

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,16 @@
 docs/_redirects
 docs/specimen/
 docs/vocabulary/
+
+# Visual Studio Code (VS Code)
+# https://github.com/github/gitignore/blob/main/Global/VisualStudioCode.gitignore
+.vscode/*
+!.vscode/settings.json
```
Comment (possumbilities) [inline .gitignore:15]: based on `line 14`, do we need `lines 15-19` ?

**Comment by possumbilities** [comment]
Fixed by:
- #318 

Closing

**Pull request closed without merging**